2022-DAEMS-001/netlogo_implementation/resourceExp.nlogo

```diff
@@ -0,0 +1,497 @@
+breed [rangers ranger] ;our agents are rangers going out from their homebase to exploit resources
+
+globals [homebase]
+
+patches-own [
+  resource ; patches have a certain value of resources
+  storage ; one of the patches is turned into a homebase where the rangers keep their stock as storage
+]
+
+rangers-own [
+  stock ;rangers can carry resources from patches to their homebase as stock
+]
+
+to setup
+  ca
+  create-rangers 1 [
+    set stock 0 ; give turtle empty stock
+    set homebase patch-here ; agent declares current patch as its homebase
+  ]
+  ask patches [
+    set resource random 100 ; give patch a certain amount of resources between 0 and specified maximum value
+    set pcolor scale-color green resource 0 100 ; color patches based on resource value
+  ]
+  reset-ticks
+end
+
+to go
+  exploit-resources
+  add-stock-to-homebase
+  tick
+end
+
+to exploit-resources
+  ask rangers [
+    let target patches with-max [resource] ; agent looks for patches with highest resources
+    move-to one-of target ; agent moves to one of the patches with the highest resources
+    set stock stock + resource
+    ask patch-here [
+      set resource 0
```

Comment:
would be nice to update patch colour as well

Reply:
Agreed, but it is not necessary for the purpose of the module. So for the sake of simplicity I wouls prefer to leave it as it is.